### DISCUSSION TITLE
New option: Add parentheses to callable variable completions

### DISCUSSION THREAD

**Discussion by MinhQuan-Luu** [microsoft/pylance-release#3537] [Enhancement]
For example: 'random.randint()' instead of just 'random.randint'
I would like an option to perform that autocomplete behaviour.

It should also add parentheses to callable classes except for type hints.
For example: 'str()', 'type()', etc.

**Comment by debonte** [comment]
Comment for triage -- If this is a common pattern, maybe `"python.analysis.completeFunctionParens": true` should be modified to do this rather than having a separate option?

[`updateInsertTextForAutoParensIfNeeded`](https://github.com/microsoft/pyrx/blob/main/packages/pylance-internal/src/server.ts#L1909) currently only acts on `CompletionItemKind.Function` and `CompletionItemKind.Method`.

**Comment by rchiodo** [comment]
What was the decided course of action here? Follow Erik's recommendation?

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
